### PR TITLE
[netcore] Create netcore specific versions of refsource classes.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: mcs/class/System.Private.CoreLib/System/GC.cs [hunk 1]
```diff
@@ -2,8 +2,244 @@
...
+
+		internal static readonly object EPHEMERON_TOMBSTONE = get_ephemeron_tombstone ();
+
+		internal static void GetMemoryInfo(out uint highMemLoadThreshold,
+												  out ulong totalPhysicalMem,
+												  out uint lastRecordedMemLoad,
+												  // The next two are size_t
+												  out UIntPtr lastRecordedHeapSize,
+												  out UIntPtr lastRecordedFragmentation)
+		{
```
Comment (filipnavara) [inline mcs/class/System.Private.CoreLib/System/GC.cs]: This API was reworked, made public and moved to shared partition:
https://github.com/dotnet/corert/commit/0227127ff610d2f4c0b8b87b38ecb40366311560

Comment (filipnavara) [reply mcs/class/System.Private.CoreLib/System/GC.cs]: I should have been clearer. There's still one non-shared method that is needed:
`public static GCMemoryInfo GetGCMemoryInfo() => default;`

Comment (vargaz) [reply mcs/class/System.Private.CoreLib/System/GC.cs]: We have that.

Comment (filipnavara) [reply mcs/class/System.Private.CoreLib/System/GC.cs]: Ah, sorry. I missed it. It was just behind the edge of the GitHub diff :)

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by filipnavara** [comment]
This breaks running tests:
```
Unhandled Exception:
Xunit.Sdk.TrueException: A test System.Runtime.Tests, Version=4.2.1.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51!System.ComponentModel.Tests.DefaultValueAttributeTests+<>c.<Ctor_TypeDescriptorNotFound_ExceptionFallback>b__5_0 forgot to Dispose() the result of RemoteInvoke()
Expected: True
Actual:   False
   at Xunit.Assert.True(Nullable`1 condition, String userMessage)
   at Xunit.Assert.True(Boolean condition, String userMessage)
   at Microsoft.DotNet.RemoteExecutor.RemoteInvokeHandle.Dispose(Boolean disposing)
   at Microsoft.DotNet.RemoteExecutor.RemoteInvokeHandle.Finalize()
[ERROR] FATAL UNHANDLED EXCEPTION: Xunit.Sdk.TrueException: A test System.Runtime.Tests, Version=4.2.1.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51!System.ComponentModel.Tests.DefaultValueAttributeTests+<>c.<Ctor_TypeDescriptorNotFound_ExceptionFallback>b__5_0 forgot to Dispose() the result of RemoteInvoke()
Expected: True
Actual:   False
   at Xunit.Assert.True(Nullable`1 condition, String userMessage)
   at Xunit.Assert.True(Boolean condition, String userMessage)
   at Microsoft.DotNet.RemoteExecutor.RemoteInvokeHandle.Dispose(Boolean disposing)
   at Microsoft.DotNet.RemoteExecutor.RemoteInvokeHandle.Finalize()
```

Looks like `GC.SuppressFinalize` is somehow broken.

**Pull request merged**